### PR TITLE
Revert "[rtl/sram] fix otp<->sram connection parameters"

### PR DESCRIPTION
**Pull request by imphil**
This reverts commit 68bae0e1ac0e47ce98e6214cf604c3af6d25234c.

This commit is presumed to break flashing application images to the
FPGA; the root cause isn't clear, and it's also likely a combination of
two commits causing this behavior.

See #5029 for further tracking.

### CODE REVIEW

File: hw/top_earlgrey/data/top_earlgrey.hjson [hunk 1]
```diff
@@ -201,7 +201,7 @@
   // generated: A module is templated and generated as part of topgen
   // top_only: A module is not templated but is specific to 'top_*' instead of 'ip'
   module: [
-    { name: "uart0",    // instance name
+    { name: "uart",     // instance name
```
Comment (tjaychen) [inline hw/top_earlgrey/data/top_earlgrey.hjson:204]: are we backing out the multiple uart commit also?

Comment (imphil) [reply hw/top_earlgrey/data/top_earlgrey.hjson:204]: Not sure yet, reverting only the first patch didn't make a difference to the FPGA test, so there's something odd going on.

**Comment by tjaychen** [comment]
there are two PRs that might be suspicious.
I recently did PRs to add multiple uarts, and another to increase sram /
flash to production sizes.
The latter is too big for the fpga, so we are relying on the reduce script
to bring things back down.

Do you want to try those too? To be fair, all those I rebased on ToT and
passed PR before merging..

**Comment by imphil** [comment]
Uh, now I understand what happened. The build that failed first was a batched commit with 7 changes in it (https://dev.azure.com/lowrisc/opentitan/_traceability/runview/changes?currentRunId=32144). I'll rebuild all of those individually to better see what happened.

**Comment by tjaychen** [comment]
so where i thought i saw issues was the fifo rst behavior one.  And i have
a PR to back that out.
but it seemed like with that one in, the failures weren't consistent.

**Comment by tjaychen** [comment]
just another note, when I first added the multiple uart PR, it DID cause
FPGA to fail.
This was happening mostly because the software ordering of pins and the
hardware ordering of pins got out of wack, and the boot ROM didn't know it
was supposed to enter boot strap.
So the whole thing hung.

I SHOULD have fixed that, but it's hard to say if that fix combined with
another commit didn't cause some kind of separate issue where things are
STILL out of sync.

**Comment by imphil** [comment]
We'll just need a bit more time here to know what's exactly going on, let's see. I'll get back as soon as the jobs finish.

**Comment by imphil** [comment]
This backout is clearly not the right one, please follow https://github.com/lowRISC/opentitan/issues/5029 for how this evolves.

**Pull request closed without merging**